gateway/1.5/modules/ROOT/pages/policies-pdk-template-caching-policy.adoc

```diff
@@ -0,0 +1,424 @@
+                logger::warn!("Unexpected error calculating cache expiration time.")
+            }
+        }
+    }
+}
+
+// Policy Configuration
+#[entrypoint]
+async fn configure(
+    launcher: Launcher,
+    Configuration(bytes): Configuration,
+    cache_builder: CacheBuilder,
+) -> Result<()> {
+    // Deserialize the configuration
+    let config: Config = serde_json::from_slice(&bytes).unwrap();
+
+    // Create the cache
+    let cache = cache_builder.build(
+        "awesome-caching".to_string(),
+        config.max_cached_values as usize,
+    );
+
+    let filter = on_request(|request_state| request_filter(request_state, &config, &cache))
+        .on_response(|response_state, request_data| {
+            response_filter(response_state, request_data, &config, &cache)
+        });
+
+    launcher.launch(filter).await?;
+    Ok(())
+}
+----
+====
+
+. Read the function documentation provided in the Rust file.
+. Edit `src/lib.rs` to fit the needs of your policy.
+
+== Test the Policy
+
+To test the policy:
+
+. Run the `build` command to compile the policy:
++
+[source,ssh]
+----
+make build
+----
++
+For more information about compiling policies, see xref:policies-pdk-compile-policies.adoc[].
+. Configure the `test/config/api.yaml` as follows:
++
+.Expand to view the `yaml` configuration example.
+[%collapsible]
+====
+[source,yaml]
+----
+# Copyright 2023 Salesforce, Inc. All rights reserved.
+---
+apiVersion: gateway.mulesoft.com/v1alpha1
+kind: ApiInstance
+metadata:
+  name: ingress-http
+spec:
+  address: http://0.0.0.0:8081
+  services:
+    upstream:
+      address: http://backend
+      routes:
+        - config:
+            destinationPath: /anything/echo/
+  policies:
+    - policyRef:
+        name: awesome-caching-v1-0-impl
+      config:
+        max_cached_values: 10
+        start_hour: 18
+        end_hour: 10
+----
+====
+
+. Configure a Flex Gateway instance to debug the policy.
++
+For more information, see xref:policies-pdk-debug-local.adoc[].
+
+. Run the `run` command to start the Flex Gateway instance:
++
+[source,ssh]
+----
+make run
+----
++
+NOTE: The `run` command starts a Flex Gateway instance inside a Docker container. By default, Docker containers use UTC as the local time. 
+
+. Send a request to Flex Gateway:
++
+[source,ssh]
+----
+curl http://127.0.0.1:8081/catalog/1 -H "cache_check: cache_value"
+----
++
+NOTE: The upstream service used in the debugging environment included with PDK responds with an echo of the request.
+
+. Send another request to Flex Gateway changing the included header:
++
+[source,ssh]
+----
+curl http://127.0.0.1:8081/catalog/1 -H "cache_check: cache_value1"
+----
++
+If the requests were made outside of business hours, the header `cache_check: cache_value` is included in the response body instead of `cache_check: cache_value1`. If the requests were made inside of business hours, the header `cache_check: cache_value1` is included in the response body.
```

Comment:
```suggestion
If requests are made outside of business hours, the header `cache_check: cache_value` is included in the response body instead of `cache_check: cache_value1`. If requests are made inside of business hours, the header `cache_check: cache_value1` is included in the response body.
```

Reply:
Logic: Avoid past tense